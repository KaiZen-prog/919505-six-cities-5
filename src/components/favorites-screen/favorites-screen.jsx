import React, {useEffect} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {fetchFavoriteOffers} from '../../store/api-actions';
import MainHeader from "../main-header/main-header";
import MainFooter from "../main-footer/main-footer";
import FavoritesLoaded from "../favorites-loaded/favotites-loaded";
import FavoritesNotLoaded from "../favorites-not-loaded/favorites-not-loaded";

const FavoritesScreen = (props) => {
  const {favoriteOffers, isFavoriteOffersLoaded, fetchFavoriteOffersAction} = props;

  useEffect(() => {
    fetchFavoriteOffersAction();
  }, []);

  const isZeroOffersToShow = favoriteOffers.length === 0 || isFavoriteOffersLoaded === false;

  return (
    <div className={`page ${(isZeroOffersToShow) ? `page--favorites-empty` : ``}`}>
      <MainHeader/>
      {isZeroOffersToShow
        ? <FavoritesNotLoaded/>
        : <FavoritesLoaded/>
      }
      <MainFooter/>
    </div>
  );
};

FavoritesScreen.propTypes = {
  favoriteOffers: PropTypes.array.isRequired,
  isFavoriteOffersLoaded: PropTypes.bool.isRequired,
  fetchFavoriteOffersAction: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  fetchFavoriteOffersAction: () => dispatch(fetchFavoriteOffers())
});

const mapStateToProps = (state) => ({
  favoriteOffers: state.APP_DATA.favoriteOffers,
  isFavoriteOffersLoaded: state.APP_DATA.isFavoriteOffersLoaded
});

export {FavoritesScreen};
export default connect(mapStateToProps, mapDispatchToProps)(FavoritesScreen);
