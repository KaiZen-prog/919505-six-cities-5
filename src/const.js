export const RATING_SCALE_MULTIPLIER = 20;

export const Cities = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export const SortingTypes = {
  POPULAR: `Popular`,
  LOW_TO_HIGH: `Price: low to high`,
  HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`
};

export const CitiesContainersClasses = {
  MAIN: {
    WITH_OFFERS: `page__main page__main--index`,
    NO_OFFERS: `page__main page__main--index page__main--index-empty`
  },

  CONTAINER: {
    WITH_OFFERS: `cities__places-container container`,
    NO_OFFERS: `cities__places-container cities__places-container--empty container`
  }
};

export const OfferCardArticleClasses = {
  MAIN_SCREEN: `cities__place-card place-card`,
  OFFER_SCREEN: `near-places__card place-card`
};

export const OfferCardImgWrapperClasses = {
  MAIN_SCREEN: `cities__image-wrapper place-card__image-wrapper`,
  OFFER_SCREEN: `near-places__image-wrapper place-card__image-wrapper`
};

export const CityMapClasses = {
  MAIN_SCREEN: `cities__map map`,
  OFFER_SCREEN: `property__map map`
};

export const CityMapSettings = {
  CITY: [52.38333, 4.9],
  ZOOM: 12,
  DEFAULT_ICON_URL: `img/pin.svg`,
  ACTIVE_ICON_URL: `img/pin-active.svg`,
  ICON_SIZE: [30, 30]
};
