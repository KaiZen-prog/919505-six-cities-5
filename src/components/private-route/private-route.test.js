import React from "react";
import renderer from "react-test-renderer";
import PrivateRoute from "./private-route";
import {Router as BrowserRouter} from "react-router-dom";
import browserHistory from "../../browser-history";
import {Provider} from "react-redux";
import configureStore from 'redux-mock-store';
import {AppRoute} from "../../const";
import {FavoritesScreen} from "../favorites-screen/favorites-screen";

const mockStore = configureStore();

describe(`PrivateRoute render correctly`, () => {
  it(`PrivateRoute auth`, () => {
    const initialState = {
      USER: {
        authorizationStatus: `AUTH`
      }
    };

    const store = mockStore(initialState);

    const tree = renderer
      .create(
          <Provider store={store}>
            <BrowserRouter history={browserHistory}>
              <PrivateRoute
                authorizationStatus={`AUTH`}
                exact
                path={AppRoute.FAVORITES}
                render={() => {
                  return (
                    <FavoritesScreen/>
                  );
                }}
              />
            </BrowserRouter>
          </Provider>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`PrivateRoute noauth`, () => {
    const initialState = {
      USER: {
        authorizationStatus: `NO_AUTH`
      }
    };

    const store = mockStore(initialState);

    const tree = renderer
      .create(
          <Provider store={store}>
            <BrowserRouter history={browserHistory}>
              <PrivateRoute
                authorizationStatus={`NO_AUTH`}
                exact
                path={AppRoute.FAVORITES}
                render={() => {
                  return (
                    <FavoritesScreen/>
                  );
                }}
              />
            </BrowserRouter>
          </Provider>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

