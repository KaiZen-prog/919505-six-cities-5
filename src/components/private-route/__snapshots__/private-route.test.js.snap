// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute render correctly PrivateRoute auth 1`] = `
<div>
  Private test
</div>
`;

exports[`PrivateRoute render correctly PrivateRoute noauth 1`] = `null`;
